tables_io/io_layer.py

```diff
@@ -9,24 +10,28 @@
     from astropy.io import fits
     HAS_ASTROPY = True
 except ImportError:  #pragma: no cover
+    print("Warning: astropy.table is not available.  tables_io will still work with other formats.")
```

Comment:
I tend to find it more convenient to leave out these warnings and just raise errors when actually calling the functions that won't work, especially for the slightly less common libs like pyarrow.

Reply:
So, @jchiang87 requested that the code not just proceed until it crashes.    In conjunction with the next comment, I think that I will 1. implement the lazy import and 2. have the default be to warn if modules aren't available, but make that easy to silence.